pkg/systemd/overview-cards/smart-status.jsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.js";
+import { ExclamationCircleIcon } from "@patternfly/react-icons";
+import { Flex } from "@patternfly/react-core/dist/esm/layouts/Flex/index.js";
+import { Icon } from '@patternfly/react-core/dist/esm/components/Icon/index.js';
+
+import cockpit from "cockpit";
+import { useEvent, useInit } from 'hooks';
+
+function countFailingDisks(proxies) {
+    const sataFail = Object.keys(proxies.drives_ata).reduce((acc, drive) => {
+        const smart = proxies.drives_ata[drive];
+        if (smart.SmartFailing) {
+            return acc + 1;
+        } else {
+            return acc;
+        }
+    }, 0);
+
+    const nvmeFail = Object.keys(proxies.nvme_controller).reduce((acc, drive) => {
+        const smart = proxies.nvme_controller[drive];
+        if (smart.SmartCriticalWarning.length > 0) {
+            return acc + 1;
+        } else {
+            return acc;
+        }
+    }, 0);
+
+    return sataFail + nvmeFail;
+}
+
+const udisksdbus = cockpit.dbus("org.freedesktop.UDisks2", { superuser: "try" });
+const proxies = {
+    drives_ata: null,
+    nvme_controller: null,
+};
+
+export const SmartOverviewStatus = () => {
+    useEvent(udisksdbus, "notify");
+
+    useInit(async () => {
+        const addProxy = (iface) => {
+            return udisksdbus.proxies("org.freedesktop.UDisks2." + iface, "/org/freedesktop/UDisks2");
+        };
+
+        proxies.drives_ata = addProxy("Drive.Ata");
+        proxies.nvme_controller = addProxy("NVMe.Controller");
+        await Promise.all(Object.keys(proxies).map(proxy => proxies[proxy].wait()));
```

Comment:
Establishing proxies for this is rather expensive -- these will receive *all* property changes for the whole lifetime of the Cockpit session. Do you have some `console.log()` measurements how often this gets re-rendered? There may be a lot of notifications which have nothing to do with smart updates, like mounting/unmounting partitions. I hope/guess it's not too bad, but please ensure this doesn't get re-rendered unnecessarily too much.

Reply:
Is there a way to make it even cheaper? This is already subscribing to the lowest number of interfaces which are very specific for smart info., I dont think I can only ask for the properties that I need.

Reply:
I'll try that then 😄 I am not a dbus mastermind but since I'm only using the smart interfaces maybe that already filters out a lot of the updates

Reply:
Yup, so it used to render on basically any storage event so I changed it to only listen to the `changed` event on both interfaces. I misunderstood how the whole dbus machinery works when it comes to notifications.